raiden/network/transport/matrix/transport.py

```diff
@@ -893,6 +933,7 @@ def _get_retrier(self, receiver: Address) -> _RetryQueue:
         if receiver not in self._address_to_retrier:
             retrier = _RetryQueue(transport=self, receiver=receiver)
             self._address_to_retrier[receiver] = retrier
+            retrier.greenlet.link_exception(self.on_error)
```

Comment:
Another nice and easy to miss bug: The retriers were not exception linked to the Transport.